Marlin/Marlin_main.cpp

```diff
@@ -6077,18 +6115,23 @@ inline void gcode_M109() {
     #define TEMP_CONDITIONS (wants_to_cool ? thermalManager.isCoolingHotend(target_extruder) : thermalManager.isHeatingHotend(target_extruder))
   #endif
 
-  float theTarget = -1.0, old_temp = 9999.0;
+  float target_temp = -1.0, old_temp = 9999.0;
   bool wants_to_cool = false;
   wait_for_heatup = true;
   millis_t now, next_temp_ms = 0, next_cool_check_ms = 0;
 
   KEEPALIVE_STATE(NOT_BUSY);
 
+  #if ENABLED(PRINTER_EVENT_LEDS)
+    const float start_temp = thermalManager.degTargetHotend(target_extruder);
+    uint8_t old_blue = 0;
```

Comment:
Ah, right… `thermalManager.degHotend` is better!

Since `old_blue` is altered below, it can't be `const`. But it should be 255 to start, because this ensures the color will be set on the first iteration.